File: bindings/csharp/FastParse/FastParseException.cs
namespace FastParse;

/// <summary>
/// Exception thrown when the native FastParse call fails.
/// </summary>
public sealed class FastParseException : Exception
{
    /// <summary>
    /// Creates a native parse exception.
    /// </summary>
    /// <param name="status">Status returned by the exported C function.</param>
    /// <param name="nativeStatus">Status stored in the native result.</param>
    /// <param name="message">Native error detail.</param>
    public FastParseException(int status, int nativeStatus, string message)
        : base($"FastParse failed with status {status}/{nativeStatus}: {message}")
    {
        Status = status;
        NativeStatus = nativeStatus;
    }

    /// <summary>Status returned by the exported C function.</summary>
    public int Status { get; }

    /// <summary>Status stored in the native result.</summary>
    public int NativeStatus { get; }
}

File: bindings/java/fastparse/src/main/java/dev/fastparse/FastParseException.java
package dev.fastparse;

/** Exception thrown when FastParse native loading or native calls fail. */
public class FastParseException extends RuntimeException {
    private final int apiStatus;
    private final int nativeStatus;

    public FastParseException(String message) {
        this(0, 0, message);
    }

    public FastParseException(int apiStatus, int nativeStatus, String message) {
        super(message);
        this.apiStatus = apiStatus;
        this.nativeStatus = nativeStatus;
    }

    public int getApiStatus() { return apiStatus; }
    public int getNativeStatus() { return nativeStatus; }
}
